cmd/celestia-appd/cmd/app_server.go

```diff
@@ -43,12 +46,27 @@ func NewAppServer(logger log.Logger, db dbm.DB, traceStore io.Writer, appOptions
 		panic(err)
 	}
 
+	// Try to determine the app version or use latest
+	appVersion := getAppVersion(logger, db)
+
+	// Create the appropriate encoding configuration based on the app version
+	var encodingConfig encoding.Config
+	if appVersion < v3.Version {
+		// For versions 1 and 2, use the standard encoding config without recursion limits
+		logger.Info("Using standard encoding config without recursion limits", "app_version", appVersion)
+		encodingConfig = encoding.MakeConfig(app.ModuleEncodingRegisters...)
+	} else {
+		// For version 3+, use the versioned encoding config with recursion limits
+		logger.Info("Using versioned encoding config with recursion limits", "app_version", appVersion)
+		encodingConfig = encoding.MakeVersionedConfig(appVersion, app.ModuleEncodingRegisters...)
+	}
```

Comment:
would this work as expected? If I only start up once, then I would be using the encodingConfig for version 1 even if I eventually sync up to a height with app version 3 or greater

Reply:
I had the same concern. This needs testing